modules/search/class.jetpack-search.php

```diff
@@ -148,6 +181,7 @@ public function search( array $es_args ) {
 		$response_code = wp_remote_retrieve_response_code( $request );
 
 		if ( ! $response_code || $response_code < 200 || $response_code >= 300 ) {
+			do_action( 'failed_jetpack_search_query', $response_code );
```

Comment:
I think we should pass in the full error so we can also output the error message and aid debugging. Maybe also pass through the request body so we can eventually add support for the debug bar plugin (we have this on wp.com so it is just porting it).

Reply:
Done